hudi-common/src/main/avro/HoodieMetadata.avsc

```diff
@@ -427,6 +427,15 @@
                             "type": "int",
                             "default": 0,
                             "doc": "Represents fileId encoding. Possible values are 0 and 1. O represents UUID based fileID, and 1 represents raw string format of the fileId. \nWhen the encoding is 0, reader can deduce fileID from fileIdLowBits, fileIdHighBits and fileIndex."
+                        },
+                        {
+                            "name": "position",
+                            "type": [
+                                "null",
+                                "long"
+                            ],
+                            "default": null,
```

Comment:
Should we instead have adefault of `-1L`? cc @yihua @nsivabalan 
That's what we use in `HoodieRecordLocation` - https://github.com/apache/hudi/blob/9f0130442a502bff6d6f7a649a7808a03d51da41/hudi-common/src/main/java/org/apache/hudi/common/model/HoodieRecordLocation.java#L46

Reply:
I think `null` is fine here for compatible schema.  When converting between Hudi metadata and in-memory position, we can convert `null` to `-1L`, or skip the record position-related logic.